src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunLibraryRootProvider.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.TypeSystem.Ecma;
+using Internal.TypeSystem;
+
+namespace ILCompiler
+{
+    /// <summary>
+    /// Provides compilation group for a library that compiles everything in the input IL module.
+    /// </summary>
+    public class ReadyToRunLibraryRootProvider : ICompilationRootProvider
+    {
+        private EcmaModule _module;
+
+        public ReadyToRunLibraryRootProvider(EcmaModule module)
+        {
+            _module = module;
+        }
+
+        public void AddCompilationRoots(IRootingServiceProvider rootProvider)
+        {
+            foreach (TypeDesc type in _module.GetAllTypes())
+            {
+                try
+                {
+                    rootProvider.AddCompilationRoot(type, "Library module type");
+                }
+                catch (TypeSystemException)
+                {
+                    // TODO: fail compilation if a switch was passed
```

Comment:
Let's just remove these TODOs in this new file since we're obviously not getting to them any time soon 😂